keeper-secrets-manager/operations.py

```diff
@@ -0,0 +1,156 @@
+                self.secrets_manager = SecretsManager(
+                    config=FileKeyValueStorage(file_name),
+                    verify_ssl_certs=self.verify_ssl
+                )
+                connector_info.update({'secrets_manager': self.secrets_manager})
+
+        except Exception as e:
+            raise ConnectorError("Failed to initialize KSM Client. " + str(e))
+        self.headers = {'Content-Type': 'application/json'}
+
+    def get_password_details(self, config, connector_info):
+        try:
+            records = connector_info.get('secrets_manager').get_secrets() or []
+            return records
+        except Exception as err:
+            logger.exception("{0}".format(str(err)))
+            raise ConnectorError("{0}".format(str(err)))
+
+    def get_credentials_inner(self, config, params, connector_info):
+        records, formatted_output = [], []
+        all_secrets = connector_info.get('secrets_manager').get_secrets() or []
+        for sec in all_secrets:
+            records.append(sec.dict)
+        if records:
+            for r in records:
+                if r.get('type') == "login":
+                    formatted_output.append(
+                        {
+                            "key": r.get('title'),
+                            "display_name": r.get('title')
+                        }
+                    )
+        return formatted_output
+
+    def get_credentials_details_inner(self, config, params, connector_info):
+        formatted_output = []
+        Object = params.get('secret_id')
+        params.update({'title': Object})
+        secret = connector_info.get('secrets_manager').get_secret_by_title(Object) or []
+        if secret:
+            records = secret.dict
+            if records:
+                for field in records.get('fields'):
+                    formatted_output.append(
+                        {
+                            "field_name": field.get('type'),
+                            "value": "*****"
+                        }
+                    )
+        return formatted_output
+
+    def get_credential_inner(self, config, params, connector_info):
+        Object = params.get('secret_id')
+        attribute_name = params.get('attribute_name')
+        secret = self.secrets_manager.get_secret_by_title(Object) or []
+        if secret:
+            records = secret.dict
+            if records:
+                for field in records.get('fields'):
+                    if field.get('type') == attribute_name:
+                        return {
+                            "password": field.get('value')[0]
+                        }
+                    else:
+                        continue
+                return {
+                    "message": "Invalid Attribute"
+                }
+
+
+def get_password(config, params, connector_info):
+    try:
+        records = []
+        ksm = KeepersSecretManager(config, connector_info=connector_info, flag=False)
```

Comment:
Instead of passing False value to flag every time its default value can be set in constructor.